lib/ui/window.dart

```diff
@@ -179,6 +185,30 @@ class Window {
   /// you can then obtain a [Scene] object, which you can display to
   /// the user via this [render] function.
   void render(Scene scene) native "Window_render";
+
+  /// Whether the user has requested that [updateSemantics] be called when
+  /// the semantic contents of window changes.
+  ///
+  /// The [onSemanticsEnabledChanged] callback is called whenever this value
+  /// changes.
+  bool get semanticsEnabled => _semanticsEnabled;
+  bool _semanticsEnabled = false;
+
+  /// A callback that is invoked when the value of [semanticsEnabled] changes.
+  VoidCallback onSemanticsEnabledChanged;
+
+  /// A callback that is invoked whenever the user requests an action to be
+  /// performed.
+  ///
+  /// This callback is used when the user expresses the action they wish to
+  /// perform based on the semantics supplied by [updateSemantics].
+  SemanticsActionCallback onSemanticsAction;
+
+  /// Change the retained semantics data about this window.
+  ///
+  /// If [semanticsEnabled] is true, the user has requested that this funciton
+  /// be called whenever the semantic content of this window changes.
```

Comment:
does this dispose the update for you or do you then have to update it?

document the answer